### DISCUSSION TITLE
v3.0.x

### DISCUSSION THREAD

**Discussion by bjlittle** [SciTools/iris#4071] [Releases]
The release manager for `v3.0.x` is @bjlittle 

The release feature branch is [v3.0.x](https://github.com/SciTools/iris/tree/v3.0.x)

| Major Release | Status | Milestone | Project | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v3.0.0rc0` | Completed | [v3.0.0](https://github.com/SciTools/iris/milestone/37) | [Iris v3.0.0](https://github.com/SciTools/iris/projects/23) | 02/10/2020 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.0.0rc0/whatsnew/3.0.html) | [v3.0.0rc0](https://github.com/SciTools/iris/releases/tag/v3.0.0rc0) | [Available](https://anaconda.org/conda-forge/iris?version=3.0.0rc0&channel=rc_iris) | - |
| `v3.0.0` | Completed | [v3.0.0](https://github.com/SciTools/iris/milestone/37) | [Iris v3.0.0](https://github.com/SciTools/iris/projects/23) | 25/01/2021 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.0.0/whatsnew/3.0.html) | [v3.0.0](https://github.com/SciTools/iris/releases/tag/v3.0.0) | [Available](https://anaconda.org/conda-forge/iris?version=3.0.0&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.0.0/) |

| Patch Release | Status | Milestone | Project | Date | RTD | Release | conda-forage | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v3.0.1` | Completed | - | [Iris v3.0.x](https://github.com/SciTools/iris/projects/30) | 27/01/2021 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.0.1/whatsnew/3.0.1.html) | [v3.0.1](https://github.com/SciTools/iris/releases/tag/v3.0.1) | [Available](https://anaconda.org/conda-forge/iris?version=3.0.1&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.0.1/) |
| `v3.0.2` | Completed | - | [Iris v3.0.x](https://github.com/SciTools/iris/projects/30) | 27/05/2021 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.0.2/whatsnew/3.0.2.html) | [v3.0.2](https://github.com/SciTools/iris/releases/tag/v3.0.2) | [Available](https://anaconda.org/conda-forge/iris?version=3.0.2&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.0.2/) |
| `v3.0.3` | Completed | - | [Iris v3.0.x](https://github.com/SciTools/iris/projects/30) | 07/07/2021 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.0.3/whatsnew/3.0.3.html) | [v3.0.3](https://github.com/SciTools/iris/releases/tag/v3.0.3) | [Available](https://anaconda.org/conda-forge/iris?version=3.0.3&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.0.3/) |
| `v3.0.4` | Completed | - | [Iris v3.0.x](https://github.com/SciTools/iris/projects/30) | 22/07/2021 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.0.4/whatsnew/3.0.4.html) | [v3.0.4](https://github.com/SciTools/iris/releases/tag/v3.0.4) | [Available](https://anaconda.org/conda-forge/iris?version=3.0.4&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.0.4/) |

**Comment by jamesp** [comment]
This week looks like a good week for a push on getting the iris 3.0.2 release cut, @bjlittle let us know what you need from the rest of us @SciTools/iris-devs to do so 👍

**Comment by bjlittle** [comment]
@SciTools/iris-devs I'm looking for devs to review and merge the following pull-requests from @rcomer:
* https://github.com/SciTools/iris/pull/4127, and
* https://github.com/SciTools/iris/pull/4059

into the `v3.0.x` release feature branch as part of the pending `iris` v3.0.2 release.

Anyone with spare capacity willing to take these on? :pray:

**Reply by bjlittle** [reply]
I'm going to be focusing on https://github.com/SciTools/iris/issues/4000, which requires to be addressed as a major part of the v3.0.2 patch release.

**Reply by rcomer** [reply]
I think we first need someone to merge all of `master`'s recent getting-the-tests-running commits into `v3.0.x`.  BTW I have now incorporated #4127 into #4128, which targets `master`, as that seemed the easiest way to demonstrate that the fix actually works and the tests pass.

**Reply by bjlittle** [reply]
@rcomer Cool, thanks for the update

We may also need to address `cirrus-ci` on the v3.0.x release feature branch... but we'll cross that bridge when come to it.

**Reply by trexfeathers** [reply]
I merged #4128 before reading my emails 😂 So that half is done.

Need to focus on some other work this afternoon but if no one else has picked this up by tomorrow I'll try and get my head round #4059.

**Reply by rcomer** [reply]
@jamesp already had a look at #4059 and I _think_ the main blocker is the CI issues.

@trexfeathers you need a break from reviewing!

**Reply by jamesp** [reply]
Yep I'm happy with #4059 to go in now without the tests passing, or wait if we are going to cherry pick the CI changes into the 3.0.x branch.

**Comment by rcomer** [comment]
I started making a [branch](https://github.com/rcomer/iris/tree/dependencies-update) to pull in the recent changes to get tests working with new dependencies.  The good news is the gallery tests pass locally with latest matplotlib & cartopy.  The bad news is I'm seeing about a gazillion cml failures in the main tests.  It looks like all the attributes are the same, but in a different order e.g.
```
-        <dimCoord bounds="[[0.0, 42077.5]]" id="fb105c19" long_name="level_height" points="[21038.8]" shape="(1,)" units="Unit('m')" value_type="float32">
+        <dimCoord id="fb105c19" long_name="level_height" units="Unit('m')" value_type="float32" shape="(1,)" points="[21038.8]" bounds="[[0.0, 42077.5]]">
```
does this ring any bells for anyone?

**Reply by jamesp** [reply]
yes it does!

**Reply by jamesp** [reply]
#3976

**Reply by rcomer** [reply]
Great - thanks @jamesp!  So should I pull those changes in or just test under python3.7 instead?  If v3.0.0 wasn't tested against py3.8 I'm thinking v3.0.2 probably doesn't need to be either?

**Reply by jamesp** [reply]
I don't think we have any py38 deps in the rest of the test framework so probably easiest to just drop 3.8 support (the docs for the 3.0.x branch explicitly only state 3.6 and 3.7 support). That said, I'm sure we have users of iris 3.0.x on python 3.8+, it's built on conda-forge to support all the way up to 3.9, so I would like to see this go into the feature branch 👍

**Reply by jamesp** [reply]
Looking at https://anaconda.org/conda-forge/iris/files?version=3.0.1 it seems that the linux-py3.8 version is the most downloaded version of iris 3.0.1

**Reply by zklaus** [reply]
Just as another data point, in ESMValTool some people have started talking about dropping Python 3.8 and only supporting Python 3.9. I'm sure that won't happen, but just goes to show that supporting different versions for a reasonable amount of time is quite important.

**Reply by rcomer** [reply]
I happened to notice recently that Cartopy are looking at adopting NEP29 for their dependency support decisions.  Perhaps Iris (and ESMValTool?) could do the same, but that's probably a discussion for another day!
https://github.com/SciTools/cartopy/issues/1518

**Reply by rcomer** [reply]
Now at #4130, but I've been blocked by the CLA checker! 😆

**Reply by rcomer** [reply]
CLA checker is objecting to the `github-actions` bot.  This is odd, because #4125 went through OK.

**Reply by jamesp** [reply]
Where does the CLA Checker code live/run?

**Reply by rcomer** [reply]
@jamesp there is documentation for the CLA checker here: https://github.com/SciTools/scitools.org.uk/wiki/Contributors-License-Agreement-processing-details

I'm not claiming to have read or understood it!

**Reply by jamesp** [reply]
Thanks @rcomer I was just going down that rabbit hole.  Oh my, heroku and webhooks...

**Reply by jamesp** [reply]
https://github.com/SciTools-incubator/scitools-cla-checker/issues/22

**Reply by bjlittle** [reply]
> CLA checker is objecting to the `github-actions` bot. This is odd, because #4125 went through OK.

@rcomer I think I've got the privileges to fix this... I recently manually injected the necessary metadata to get the `pre-commit.ci` bot to pass the CLA.

Hang tight and I'll take a peek...

**Reply by rcomer** [reply]
@bjlittle thanks, but it didn’t prevent the PR getting merged.  So not urgent as far as I’m aware.

**Comment by rcomer** [comment]
Can I interest anyone in reviewing some link repairs? #4132

**Reply by bjlittle** [reply]
Sure :smile:

**Comment by rcomer** [comment]
Does this still need merging back into `master`?

**Reply by bjlittle** [reply]
Yup, thanks... that's on me, no worries :+1:

**Reply by bjlittle** [reply]
@rcomer Thanks for the nudge :smile:

**Comment by bjlittle** [comment]
We're hoping to unpin `cftime` and make this available to the community asap as `iris` 3.0.3, rather than keeping this until an `iris` 3.1 release (which is currently scheduled for Sept '21)

There is package dependency trouble brewing between `iris`, `cftime`, `nc-time-axis` and `cf-units`... so the sooner we address this, then the better :+1:

**Reply by zklaus** [reply]
I realize this is a lot of uninteresting work. Thanks for doing it! It is really important!

**Reply by bjlittle** [reply]
@zklaus We're glad to help :smile: 

Are you guys scheduling an `esmvaltool` 2.2.1 patch release? If so, any idea when? That might help us prioritise at this end :+1:

**Reply by zklaus** [reply]
Good question. I don't think we had planned for that, but I'll bring it up.

**Reply by bouweandela** [reply]
We're not scheduling an esmvaltool 2.2.1 patch release, but @zklaus is in the process of releasing esmvaltool 2.3 (planned [this week](https://docs.esmvaltool.org/en/latest/community/release_strategy.html#release-schedule)?)

**Reply by bjlittle** [reply]
@zklaus and @bouweandela I should be able to get 3.0.3 out this week, but it also relies on a `cf-units` 3.0.1 patch release to support it, which is also in progress :+1:

HTH

**Comment by pp-mo** [comment]
Stacked up #4230 for inclusion in v3.0.4
but _not_ for 3.0.3, as we don't want any further risk or delay with that

**Comment by bjlittle** [comment]
`iris` 3.0.4 is now available. Whoop!

Thank you so much to everyone who contributed to this patch release. 

A fantastic team effort! Awesome job :tada: :pray: :clap:

**Reply by jamesp** [reply]
Great work everyone, great to see so much momentum on iris

**Comment by bjlittle** [comment]
I still need to merge back `v3.0.x` to `main`... this will include the stacked changes from 3.0.2, 3.0.3 and 3.0.4 :blush:

**Comment by bjlittle** [comment]
Post 3.0.4 release I noticed that PyPI is reporting the release as failing

![image](https://user-images.githubusercontent.com/2051656/126757202-70bbaec7-1378-42e0-890b-06e5aaa12969.png)

This is due to the badge still pointing at `master` instead of `main` (BTW release badges have always pointed to `master`/`main` instead of the feature release branch)

The tests and release are actually all good, and I propose that we **don't** fix this in a specific post patch release.

Instead, we can wait for `iris` 3.1rc0 in August and 3.1.0 in September to clobber this. I'm totally okay with that.

**Comment by pp-mo** [comment]
Archiving = recategorise as "archived"
**WAS** category = "Releases" 🚩
